server/server.go

```diff
@@ -514,6 +516,82 @@ func (s *server) recvLoop(stream rpcpb.ControlService_StreamStatusServer) error
 	}
 }
 
+func (s *server) AddNode(ctx context.Context, req *rpcpb.AddNodeRequest) (*rpcpb.AddNodeResponse, error) {
+	zap.L().Debug("received add node request", zap.String("name", req.Name))
+
+	if info := s.getClusterInfo(); info == nil {
+		return nil, ErrNotBootstrapped
+	}
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	var logLevel, whitelistedSubnets, pluginDir string
+
+	nodeName := req.Name
```

Comment:
I don't think we need this variable

Reply:
This is a debatable nit really, but ok.

Reply:
We should use one or the other but not both